src/Controls/src/Core/Handlers/Items/iOS/CarouselViewController.cs

```diff
@@ -531,28 +546,44 @@ void UpdateInitialPosition()
 
 			if (!InitialPositionSet)
 			{
-				if (ItemsView is not CarouselView carousel)
-				{
-					return;
-				}
-
-				InitialPositionSet = true;
-
 				int position = carousel.Position;
 				var currentItem = carousel.CurrentItem;
+
 				if (currentItem != null)
 				{
-					position = ItemsSource.GetIndexForItem(currentItem).Row;
-				}
-				else
-				{
-					SetCurrentItem(position);
-				}
+					// Sometimes the item could be just being removed while we navigate back to the CarouselView
+					var positionCurrentItem = ItemsSource.GetIndexForItem(currentItem).Row;
+					if (positionCurrentItem != -1)
+					{
+						position = positionCurrentItem;
+					}
+				}				
 
-				carousel.ScrollTo(position, -1, Microsoft.Maui.Controls.ScrollToPosition.Center, false);
+				await Task.Delay(100).ContinueWith((t) =>
```

Comment:
Is this the best approach ? Why 100? can we try the Collection.PerformBatchUpdates see if it helps?

Reply:
@rmarinho , I have used the similar approach used in [CarouselViewController2.cs](https://github.com/dotnet/maui/blob/ffb0db289d7c9663c1d6bcd327ab5b8b1c4b4cd1/src/Controls/src/Core/Handlers/Items2/iOS/CarouselViewController2.cs#L514). Using Collection.PerformBatchUpdates causes the page with carousel view freeze. Please verify and let me know if you have any concerns.